### DISCUSSION TITLE
PendingMs 0 and shouldPending true

### DISCUSSION THREAD

**Discussion by OrlovAlexei** [TanStack/router#2717] [Q&A]
I noticed that even with `pendingMs: 0`, shouldPending is still `true`. Is this the expected behavior?

https://github.com/TanStack/router/blob/581ebe31f00a249ef2897e618b3ef16cd9d4914e/packages/react-router/src/router.ts#L2167

**Comment by schiller-manuel** [comment]
maybe? can you please provide a complete minimal example where that matters?

**Comment by OrlovAlexei** [comment]
Looking at the code further, there are setTimeout calls with timeout=0. Could we skip such setTimeout calls (similar to how we handle Infinity timeout)?

Suggestion: We could potentially optimize by treating timeout=0 similar to timeout=Infinity, since both are edge cases. This would mean skipping the timeout set up completely when timeout=0.

**Reply by schiller-manuel** [reply]
can you come to our discord to discuss this there? https://discord.com/invite/WrRKjPJ

**Comment by OrlovAlexei** [comment]
I noticed that the Task for handling searchParams updates is quite large/complex.
https://stackblitz.com/edit/tanstack-router-fh1cv3?file=src%2Fmain.tsx
![image](https://github.com/user-attachments/assets/280bc7d1-34c5-4466-88ec-9d6daca747c9)
